### PR TITLE
CodeEditor loading property considered by the Editor

### LINKED ISSUE #11033: Bug - CodeEditor - loading component not passed to Editor
**Describe the problem**
In case loading property is defined this is not considered by the `@monaco-editor/react` Editor component, just FMPOV it is not passed through.

**How do you reproduce the problem?**

Just by using the CodeEditor component like
```
    <CodeEditor
      ...
      loading={<Spinner />}
    />
```

the spinner is not shown and `Loading..` text is shown instead

**Expected behavior**
To see whatever the component defined on `loading` property while loading the editor

**Is this issue blocking you?**
it is not

**Screenshots**
Screenshot from 2024-09-20 11-36-43

**What is your environment?**
 - OS: Fedora 40
 - Browser Firefox, chrome
 - Version -

**What is your product and what release date are you targeting?**
HCM and 25Q1

**Any other information?**
none

### PR DESCRIPTION
**Pull request by Ginxo**
**What**: Closes #11033

**Additional issues**:

### CODE REVIEW

**Comment by Ginxo** [comment]
I would like to see tests for this particular use case not being covered, I will provide as soon as we have an approval about the issue and also the proposal (this PR)

File: packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx
Comment (kmcfaul) [inline packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx:657]: `editorProps` should already allow `loading` to be passed. If we want `loading` to be a distinct property, we will need to add it to the props table.

**Review by kmcfaul — CHANGES_REQUESTED** [review]

**Review by kmcfaul — APPROVED** [review]
Ah I do see we have this on our props table already, it wasn't being passed. 

Using `editorProps` would also work, but I'm in favor of adding this method too.

File: packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx
Comment (kmcfaul) [reply packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx:657]: I missed that this already was in the props table and is unsed.

**Comment by Ginxo** [comment]
one more additional approver required, please any kind soul aroung? :pray: :wink:

**Review by tlabaj — APPROVED** [review]

**Comment by tlabaj** [comment]
@Ginxo if you need this in v5, can you please change the base branch to the `v5` branch

**Comment by Ginxo** [comment]
> @Ginxo if you need this in v5, can you please change the base branch to the `v5` branch

a [new PR for V5](https://github.com/patternfly/patternfly-react/pull/11134) already open

**Comment by tlabaj** [comment]
@Ginxo We are still in code freeze right now.  We will merge your PR once our release is live.

**Pull request merged**
